app/src/main/java/com/monkeyteam/chimpagne/viewmodels/EventViewModel.kt

```diff
@@ -260,8 +277,16 @@ class EventViewModel(
     return buildChimpagneEvent().getRole(userUID)
   }
 
-  fun getCurrentUserRole(): ChimpagneRole {
-    return getRole(accountManager.currentUserAccount?.firebaseAuthUID ?: "")
+  fun promoteGuestToStaff(uid: ChimpagneAccountUID) {
+    _uiState.value =
+        _uiState.value.copy(
+            guests = _uiState.value.guests - uid, staffs = _uiState.value.staffs + (uid to true))
+  }
+
+  fun demoteStaffToGuest(uid: ChimpagneAccountUID) {

```

Comment:
NOT ATOMIC

Reply:
Yes a much needed change, I did as requested.